src/dc_tools.rs

```diff
@@ -67,50 +67,12 @@ pub unsafe fn dc_str_replace(
     haystack: *mut *mut libc::c_char,
     needle: *const libc::c_char,
     replacement: *const libc::c_char,
-) -> libc::c_int {
-    let mut replacements: libc::c_int = 0i32;
-    let mut start_search_pos: libc::c_int = 0i32;
-    let needle_len: libc::c_int;
-    let replacement_len: libc::c_int;
-    if haystack.is_null()
-        || (*haystack).is_null()
-        || needle.is_null()
-        || *needle.offset(0isize) as libc::c_int == 0i32
-    {
-        return 0i32;
-    }
-    needle_len = strlen(needle) as libc::c_int;
-    replacement_len = (if !replacement.is_null() {
-        strlen(replacement)
-    } else {
-        0
-    }) as libc::c_int;
-    loop {
-        let mut p2: *mut libc::c_char =
-            strstr((*haystack).offset(start_search_pos as isize), needle);
-        if p2.is_null() {
-            break;
-        }
-        start_search_pos =
-            (p2.wrapping_offset_from(*haystack) + replacement_len as isize) as libc::c_int;
-        *p2 = 0i32 as libc::c_char;
-        p2 = p2.offset(needle_len as isize);
-        let new_string: *mut libc::c_char = dc_mprintf(
-            b"%s%s%s\x00" as *const u8 as *const libc::c_char,
-            *haystack,
-            if !replacement.is_null() {
-                replacement
-            } else {
-                b"\x00" as *const u8 as *const libc::c_char
-            },
-            p2,
-        );
-        free(*haystack as *mut libc::c_void);
```

Comment:
I guess you still need to free the old hackstack

Reply:
or does to_string do it already?

Reply:
Good catch, thanks.